x-pack/plugins/observability/public/pages/alerts/containers/alerts_table_t_grid/alerts_table_t_grid.tsx

```diff
@@ -171,35 +171,67 @@ function ObservabilityActions({
   }, []);
 
   const casePermissions = useGetUserCasesPermissions();
-  const event = useMemo(() => {
-    return {
-      data,
-      _id: eventId,
-      ecs: ecsData,
-    };
-  }, [data, eventId, ecsData]);
-
   const ruleId = alert.fields['kibana.alert.rule.uuid'] ?? null;
   const linkToRule = ruleId ? prepend(paths.management.ruleDetails(ruleId)) : null;
 
+  const caseAttachments: CaseAttachments = useMemo(() => {
+    return ecsData?._id
+      ? [
+          {
+            alertId: ecsData?._id ?? '',
+            index: ecsData?._index ?? '',
+            owner: observabilityFeatureId,
+            type: CommentType.alert,
+            rule: cases.helpers.getRuleIdFromEvent({ ecs: ecsData, data: data ?? [] }),
+          },
+        ]
+      : [];
+  }, [ecsData, cases.helpers, data]);
+
+  const createCaseFlyout = cases.hooks.getUseCasesAddToNewCaseFlyout({
+    attachments: caseAttachments,
+    onSuccess: async (theCase: Case) => {
+      casesToasts.showSuccessAttach(theCase);
+    },
+  });
+
+  const selectCaseModal = cases.hooks.getUseCasesAddToExistingCaseModal({
+    attachments: caseAttachments,
+    onRowClick: (theCase?: Case) => {
+      if (theCase !== undefined) {
+        casesToasts.showSuccessAttach(theCase);
+      }
+    },
+  });
+
+  const handleAddToNewCaseClick = useCallback(() => {
+    createCaseFlyout.open();
+    closeActionsPopover();
+  }, [createCaseFlyout, closeActionsPopover]);
+
+  const handleAddToExistingCaseClick = useCallback(() => {
+    selectCaseModal.open();
+    closeActionsPopover();
+  }, [closeActionsPopover, selectCaseModal]);
```

Comment:
I do not think the consumers of cases should be responsible for showing the cases toaster. This logic should be moved inside cases. I also think we can simplify this by having one hook (that uses your hooks inside) that returns the flyout and the modal (in the future we may have only the modal in which a create button will be shown right on the filters). This will make it easier for other solutions to attach data to a case and also make it easier for us to refactor internally without the need to change what we expose.

```suggestion
  const { flyoutHandler, modalHandler } = cases.hooks.getUseAttachToCase({
    attachments: caseAttachments,
  });

  const handleAddToNewCaseClick = useCallback(() => {
    flyoutHandler.open();
    closeActionsPopover();
  }, [flyoutHandler, closeActionsPopover]);

  const handleAddToExistingCaseClick = useCallback(() => {
    modalHandler.open();
    closeActionsPopover();
  }, [modalHandler, selectCaseModal]);
```

Reply:
I agree with this logic, I was just not sure about this specific detail. Should I change it as such?

Reply:
Yes, please!